Add instances for Int8-64, Word8-64 and Float

rl-king opened the pull request:
Fixes #12

ollef commented on src/Language/Haskell/To/Elm.hs:
```diff
@@ -724,6 +728,46 @@ instance HasElmDecoder Aeson.Value Int where
     "Json.Decode.int"
 
+deriving via Int instance HasElmType Int.Int8
+deriving via Int instance HasElmEncoder Aeson.Value Int.Int8
```
What happens if we pass out-of-range numbers from Elm to Haskell using this? E.g. passing a number >= 256 as an `Int8`?

rl-king replied:
The biggest concern would be the ability to create Elm values that will error once they're passed to Haskell. I'm not sure how bad that is...

### Aeson

For the `HasElmEncode Aeson.Value a` instances we'll run in to Aeson parser failures like:

`value is either floating or will cause over or underflow 1234.0` for Word8.

We're essentially going from `Elm.Int -> Aeson.Types.Number Scientific -> a` so Aeson is handling conversion to Haskell types.

### Servant

Assuming people use servant-to-elm :)

`Capture "id" Word8 :> Get '[JSON] NoContent` will not match

`QueryParam' '[Required, Strict] "id" Word8 :> Post '[JSON] NoContent`
Errors with: `error parsing query parameter id failed: out of bounds: '1234' (should be between 0 and 255)`

ollef replied:
Okay, maybe we can live with that.

Another problem which I'm starting to think is worse is that [Elm's `Int` has range `-2^53` to `2^53 - 1`](https://package.elm-lang.org/packages/elm/core/latest/Basics#Int) when targeting Javascript. So going from `Int64` and `Word64` to Elm's `Int` is lossy. And since the usecase for these kinds of types can be binary data, sometimes computer-generated (e.g. IDs, UUIDs), they can be outside of the "human range", that you might expect for a type like `Int` which doesn't have a precisely specified bit width.

rl-king replied:
Haskell `Int` can be 64-bit, the [docs](https://hackage.haskell.org/package/base-4.12.0.0/docs/Data-Int.html#t:Int) give a lower bound and following the Bounded instance source gets me to [maxInt](https://hackage.haskell.org/package/base-4.12.0.0/docs/src/GHC.Base.html#maxInt). So that's an already present "issue".

Maybe no instance for Word64? Since that's way out of JS range.

ollef replied:
Yes, that is technically true, but because it's not specified it's usually not used in a way where you come close to the limit in my experience, whereas you often do with types like `Word64`.

Yeah, removing the `Word64` and `Int64` instances seems better to me. :+1:

rl-king replied:
👍 

And, with regards to the deriving issues, just manually define the instances I guess?

ollef replied:
I think so, yeah.

rl-king commented:
I've updated the code. Also added some comments distinguish the similar looking Int code blocks, and for the other types to get things a bit consistent.

ollef commented:
Great, thanks!
